ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionReceipt.java

```diff
@@ -101,8 +101,22 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
 
     LOG.trace("Processed private transaction receipt");
 
+    int logIndexStart = 0;
+    //  TODO:  Fetch all PrivateTransactionReceipt's in block, use to compute logIndex
```

Comment:
We should only use receipts for private transactions for the same privacy group. 
PrivateStateStorage gives access to the "PrivateBlockMetadata" with the method "Optional<PrivateBlockMetadata> getPrivateBlockMetadata(Bytes32 blockHash, Bytes32 privacyGroupId);". It contains the list of transaction hashes for private transactions in this block. Using these hashes we can retrieve the private transaction receipts using "Optional<PrivateTransactionReceipt> getTransactionReceipt(Bytes32 blockHash, Bytes32 txHash);"

Reply:
Using only the private transaction from the same privacy group makes sure that every member of the privacy group get's the same result.

Reply:
@pinges Thanks for looking at it, this sounds like a good plan!